app/components/layout/keyboard_layout/keyboard_layout.js

```diff
@@ -41,52 +40,36 @@ export default class KeyboardLayout extends PureComponent {
     }
 
     onKeyboardWillHide = () => {
-        this.setState({bottom: 0});
+        this.setState({
+            keyboardHeight: 0,
+        });
     };
 
-    onKeyboardChange = (e) => {
-        if (!e) {
-            this.setState({bottom: 0});
-            return;
-        }
-
-        const {endCoordinates} = e;
-        const {height} = endCoordinates;
-
-        this.setState({bottom: height});
+    onKeyboardWillShow = (e) => {
+        this.setState({
+            keyboardHeight: e?.endCoordinates?.height || 0,
+        });
     };
 
     render() {
-        const {children, theme, ...otherProps} = this.props;
-        const style = getStyleFromTheme(theme);
+        const layoutStyle = [this.props.style, style.keyboardLayout];
```

Comment:
Is this an RN-specific thing, or can you stack style attributes as arrays in React too? Neat, regardless :)

Reply:
I haven't actually tried in regular React, but being able to pass multiple styles in an array in RN is great. The only thing here is that I'm not sure if this works if `this.props.style` itself is an array